website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+                'page': ((self.page/2) | 0) + 1,
+                'page[size]': 20
+            };
+            if (self.eventFilter) {
+                query['filter[action]'] = self.eventFilter;
+            }
+            if (reset || init) {
+                query.aggregates = 1;
+            }
+            if (!init) {
+                query['filter[date][lte]'] = self.dateEnd.toISOString();
+                query['filter[date][gte]'] = self.dateBegin.toISOString();
+            }
+            var url = $osf.apiV2Url('users/' + self.userId + '/logs/', { query : query});
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig, background: (update ? true : false)});
+            promise.then(function(result){
+                self.loading = false;
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                });
+                if (init) {
+                    self.lastDay = moment.utc(result.data[0].attributes.date);
+                    self.dateEnd = self.lastDay;
+                    var firstDay = moment.utc(result.links.meta.last_log_date);
+                    self.firstDay = ((firstDay >= self.sixMonthsAgo) ? firstDay : self.sixMonthsAgo).startOf('month');
+                    var dateBegin = moment.utc(result.data[0].attributes.date).subtract(1, 'months');
+                    self.dateBegin = ((dateBegin > self.firstDay) ? dateBegin : self.firstDay).startOf('day');
+                    if ((self.today - self.firstDay)/2629746000 < 1){
+                        self.div = self.div/4;
+                        self.formatFloat = 'Do h a';
+                        self.steps = 14;
+                        self.formatPip = 'MMM Do';
+                    } else if ((self.today - self.firstDay)/2629746000 < 3){
+                        self.div = self.div/2;
+                        self.formatFloat = 'MMM Do h';
+                        self.steps = 28;
+                        self.formatPip = 'MMM Do';
+                    } else {
+                        self.formatFloat = 'MMM Do';
+                        self.steps = 31;
+                        self.formatPip = 'MMM';
+                    }
+                }
+                if (reset){
+                    self.totalEvents = result.links.meta.total;
+                    self.eventNumbers = result.links.meta.aggregates;
+                    self.cache = [];
+                }
+                if (!init) {
+                    self.cache.push(result.data.slice(0,9));
+                    self.cache.push(result.data.slice(10,19));
+                    if (!update) {
+                        self.activityLogs(self.cache[self.page - 1]);
+                    }
+                }
+                self.lastPage = (result.links.meta.total / (result.links.meta.per_page/2) | 0) + 1;
+            });
```

Comment:
It looks like there is no error handling here. Is that correct? Can you add a logical error handling both for us and user if necessary.